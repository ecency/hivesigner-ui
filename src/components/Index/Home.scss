.HomeContainer {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin-left: auto;
  margin-right: auto;
  grid-gap: 10rem;
  padding: 4.5rem 2rem;
  align-items: center;
  position: relative;
  transition: all 0.85s;
  @media screen and (max-width: 1280px) {
    flex-direction: column;
    margin-top: 10vh;
    grid-gap: 0 !important;
  }
  .Menubar {
    font-size: 2rem;
    color: black;
    position: absolute;
    right: 30px;
    top: -50px;
    display: block;
    @media screen and (min-width: 630px) {
      display: none;
    }
  }
  .HomeDiv {
    display: flex;
    flex-direction: column;
    z-index: -1;
    max-width: 500px;
    margin: 50px auto 0 auto;
    padding-left: 20px;
    // img {
    //   max-width: 492px;
    // }
    @media screen and (max-width: 1280px) {
      margin-top: 50px;
      padding-left: 0px !important;
    }
    @media screen and (max-width: 630px) {
      padding: 0 10px;
      * {
        width: 100%;
      }
      // img {
      //   max-width: 242px;
      // }
    }
  }
  .HomeTitle {
    font-size: 3rem;
    font-weight: 300;
    color: rgb(189, 189, 189);
    @media screen and (max-width: 1280px) {
      text-align: center;
      margin-bottom: 25px;
    }
    @media screen and (max-width: 630px) {
      display: none;
    }
  }
  .HomeDetail {
    display: flex;
    flex-direction: column;
    /* justify-content: center;
    align-items: center; */
    justify-content: space-between;
    margin-top: 30px;
    @media screen and (max-width: 630px) {
      padding: 0 10px;
      align-items: center;
      text-align: center;
      margin-top: -100px;
    }
    @media screen and (max-width: 390px) {
      margin-top: -150px;
    }
  }
  .Header {
    display: flex;
    flex-direction: row;
    align-items: center;
    font-size: 3rem;
    font-weight: 600;
    padding-bottom: 5rem;
    img {
      margin-right: 10px;
    }
    @media screen and (max-width: 1280px) {
      text-align: center;
      justify-content: center;
      font-size: 2.25rem;
      position: absolute;
      top: 0;
      margin: 2.25rem auto;
      width: 100%;
    }
    @media screen and (max-width: 350px) {
      font-size: 2rem;
    }
  }
  .DetailContent {
    font-size: 1.25rem;
    padding-bottom: 3rem;
    color: rgb(130, 130, 130);
    max-width: 600px;
    @media screen and (max-width: 1280px) {
      text-align: center;
    }
    @media screen and (max-width: 630px) {
      padding: 100px 10px 30px 10px;
      align-items: center;
      text-align: center;
      font-size: 1rem;
    }
  }
}
.Button {
  background-color: #e31337;
  color: #fff;
  padding-top: 1rem;
  padding-bottom: 1rem;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  font-size: 1.125rem;
  border-radius: 0.375rem;
  align-items: center;
  max-width: 170px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  cursor: pointer;
  margin-bottom: 50px;
  transition: all 0.85s;
  img {
    margin-right: 10px;
  }
  &:hover {
    background-color: transparent;
    color: #e31337;
    border: 1px solid #e31337;
  }
  @media screen and (max-width: 1280px) {
    margin: auto;
  }
}
.homefix {
  margin-top: 1rem;
  @media screen and (max-width: 1280px) {
    margin-top: 0;
  }
}
