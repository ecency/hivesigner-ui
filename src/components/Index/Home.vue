<template>
  <div class="hero height-full">
    <Center>
      <span class="logo iconfont icon-hivesigner" />
      <h4 class="m-0">
        hivesigner
      </h4>
      <div class="mt-4 mb-7">
        <h2 class="mb-6">
          You're in control of your keys.
        </h2>
        <div class="mb-6">
          <router-link
            to="/login?redirect=accounts"
            class="rounded-2 px-3 py-3 bg-primary text-white button-link"
          >
            <span class="iconfont icon-shield" />
            Get started
          </router-link>
        </div>
      </div>
      <Footer />
    </Center>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'nuxt-property-decorator'

@Component
export default class Home extends Vue {
}
</script>
