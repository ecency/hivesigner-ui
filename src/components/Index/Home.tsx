import { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import { useTranslation } from "react-i18next";

import { ImMenu } from "react-icons/im";
import { AiOutlineClose } from "react-icons/ai";

import Navbar from "./Navbar";
import Content from "../../layouts/content";

import HomeImage from "../../assets/img/home";
import HomeLogo from "../../assets/img/logo";
import Lock from "../../assets/img/lock";
import "./Home.scss";
const Home = () => {
  const { t } = useTranslation();
  const [showNavbar, setshowNavbar] = useState(true);
  const handleScreenChange = (mediaQuery: any) => {
    if (mediaQuery.matches) {
      setshowNavbar(true);
    } else {
      setshowNavbar(false);
    }
  };
  useEffect(() => {
    let window_width = window.matchMedia("(min-width:632px)");
    window_width.addListener(handleScreenChange);
    handleScreenChange(window_width);
    return () => {
      window_width.removeListener(handleScreenChange);
    };
  }, []);

  const handleChange = () => {
    setshowNavbar(!showNavbar);
  };

  return (
    <Content>
      <div className="HomeContainer">
        <div className="Menubar" onClick={() => handleChange()}>
          {showNavbar ? <AiOutlineClose /> : <ImMenu />}
        </div>
        <div className="HomeDiv">
          <HomeImage />
          <div className="HomeTitle">{t("index.secure_way_sign_in")}</div>
        </div>
        <div className="HomeDetail">
          <div className="Header">
            <HomeLogo />
            Hivesigner
          </div>

          <div className="DetailContent">{t("index.description")}</div>
          <Link to="/import">
            <div className="Button">
              <Lock />
              {t("index.get_started")}
            </div>
          </Link>
          {showNavbar && (
            <div className="homefix">
              <Navbar />
            </div>
          )}
        </div>
      </div>
    </Content>
  );
};
export default Home;
