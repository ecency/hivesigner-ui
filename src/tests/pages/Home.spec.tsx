import React from "react";
import { shallow } from "enzyme";
import Home from "../../components/Index/Home";
describe("Home component Test", () => {
  test("UI Test", () => {
    const home = shallow(<Home />);
    expect(home.find("DetailContent").text()).toBe(
      "Secure way to sign with Hivesigner. Best security for users and developers to integrate industry standard OAuth2 for their Blockchain applications. Transform web 2.0 apps into web 3.0 decentralized apps."
    );
  });
});
