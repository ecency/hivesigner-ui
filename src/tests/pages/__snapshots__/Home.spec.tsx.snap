// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Test Snapshot testing 1`] = `
<div
  className="HomeDetail"
/>
`;

exports[`Home component Test Snapshot testing 2`] = `
<div
  className="Menubar"
/>
`;
